tests/lib/alertmanager_api/test_schedule_silence.py

```diff
@@ -0,0 +1,198 @@
+    class_instance, missing_end_time
+):  #  pylint: disable=redefined-outer-name
+    with patch.object(class_instance.log, "critical") as mock_critical:
+        with pytest.raises(
+            MissingMandatoryParamError, match="Missing silence end time"
+        ):
+            class_instance.schedule_silence(missing_end_time)
+        mock_critical.assert_called_once_with("Missing silence end time")
+
+
+def test_missing_attribute_author(
+    class_instance, missing_author
+):  #  pylint: disable=redefined-outer-name
+    with patch.object(class_instance.log, "critical") as mock_critical:
+        with pytest.raises(MissingMandatoryParamError, match="Missing silence author"):
+            class_instance.schedule_silence(missing_author)
+        mock_critical.assert_called_once_with("Missing silence author")
+
+
+def test_missing_attribute_comment(
+    class_instance, missing_comment
+):  #  pylint: disable=redefined-outer-name
+    with patch.object(class_instance.log, "critical") as mock_critical:
+        with pytest.raises(MissingMandatoryParamError, match="Missing silence comment"):
+            class_instance.schedule_silence(missing_comment)
+        mock_critical.assert_called_once_with("Missing silence comment")
+
+
+@patch("alertmanager_api.silence.requests.post")
+def test_schedule_silence_sucess(mock_post):
+    """
+    use case: the call to schedule_silence() works fine
+    """
+    alertmanager_account = MagicMock()
+    class_instance = AlertManagerAPI(  #  pylint: disable=redefined-outer-name
+        alertmanager_account
+    )
+    date_start = datetime(2025, 1, 1, 10, 0, 0)
+    date_end = datetime(2025, 1, 1, 10, 0, 0)
+    silence = SilenceDetails("hostname", date_start, date_end, "author", "comment")
+    mock_response = MagicMock()
+    mock_response.status_code = 200
+    mock_post.return_value = mock_response
+    class_instance.schedule_silence(silence)
+    mock_post.assert_called_once()
+
+
+@patch("alertmanager_api.silence.requests.post")
+def test_schedule_silence_exception_is_raised_status_code_not_200(mock_post):
+    """
+    use case: the call to requests to schedule a silence
+              returned a RC different than 200
+    """
+    alertmanager_account = MagicMock()
+    class_instance = AlertManagerAPI(  #  pylint: disable=redefined-outer-name
+        alertmanager_account
+    )
+    date_start = datetime(2025, 1, 1, 10, 0, 0)
+    date_end = datetime(2025, 1, 1, 10, 0, 0)
+    silence = SilenceDetails("hostname", date_start, date_end, "author", "comment")
+    mock_response = MagicMock()
+    mock_response.status_code = 201
+    mock_response.text = "rc is 201"
+    mock_post.return_value = mock_response
+    try:
+        class_instance.schedule_silence(silence)
+    except RequestException as req_ex:
+        assert "Failed to create silence. " in str(req_ex)
+        assert "Response status code: 201" in str(req_ex)
+        assert "Response text: rc is 201" in str(req_ex)
```

Comment:
This test is looking more at the specific message instead of checking whether the exception is being raised in the first place in the method. It might be better for testing whether errors are return using `pytest.raises([exceptiontype])` to test the error is raised